lib/npg_pipeline/function/product_delivery_notifier.pm

```diff
@@ -154,7 +155,20 @@ sub create {
     next if not $self->is_for_s3_release_notification($product);
 
     my $msg_file = $product->file_path($self->message_dir, ext => 'msg.json');
-    $self->_write_message_file($msg_file, $self->make_message($product));
+    if (-e $msg_file) {
+      $self->info("Message file '$msg_file' exists");
+      if ($self->_message_content_changed($msg_file,
+                                          $self->make_message($product))) {
+        $self->info("Message content in file '$msg_file' has ",
+                    'changed. Writing new contents');
+        $self->_write_message_file($msg_file, $self->make_message($product));
```

Comment:
Could calling make_message() twice be more expensive than storing its result in a variable? If so, just call it once?